streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1081,6 +1081,11 @@ private void addToResetList(final TopicPartition partition, final Set<TopicParti
         partitions.add(partition);
     }
 
+    // This method is added for usage in tests where mocking the underlying native call is not possible.
+    public boolean isThreadAlive() {
```

Comment:
Note that adding this public method does not require a KIP because this class is not part of the public API at https://kafka.apache.org/32/javadoc/allclasses-index.html